subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -435,6 +996,25 @@ static int fp_storage_ak_init(void)
 		return 0;
 	}
 
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(bt_request_cb);
+	}
+
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		FP_BONDS_FOREACH(bond) {
+			/* If Fast Pair was disabled before the peer that performed the unsuccessful
+			 * Fast Pair Procedure disconnected, there may be a conn_ctx not set to
+			 * NULL. All conn_ctx should be set to NULL at this point.
+			 */
+			bond->conn_ctx = NULL;
```

Comment:
Shouldn't we somehow ensure that this incomplete bond record will be removed? Keep in mind that the same peer might have bonded without following Fast Pair flow after Fast Pair was disabled (unless I missed something, it would not be properly removed if the interrupted Fast Pair Procedure was related to subsequent pairing)

Reply:
I'm afraid it's really hard to do something about it. It's not really severe. Worst that can happen is that on AK removal we remove some bond that started subsequent pairing procedure when Fast Pair was enabled but finished the procedure when Fast Pair was disabled.

Reply:
I can create task to improve it in the future.

Reply:
Discussed offline. The edge case will be documented and @alstrzebonski will open a follow-up ticket to fix it in the future (proper fix may require tracking Fast Pair BLE bonds also while Fast Pair is disabled - initially we wanted to avoid that but it may be worth reconsidering this approach as we may be unable to prevent some of the edge cases without it).